src/js/search-bar/search-bar.view.js

```diff
@@ -25,22 +24,35 @@ export class SearchBarView extends AbstractView {
 
     afterViewInit() {
         super.afterViewInit();
-        this.searchBarOverlay = selectDomElement('#search-bar-overlay');
+        this.searchBar = selectDomElement('#search-bar');
         this.searchBtn = selectDomElement('#search-btn');
 
-        document.onkeyup = (event) => {
-            if (event.key === 'Enter') {
+        selectDomElement('#search-input').onkeyup = (event) => {
+            if(event.key === 'Enter') {
                 this.sendEvent(SearchBarEvent.CLICK_SEARCH);
-            };
-        };
-        this.searchBarOverlay.onclick = () => { 
-            this.sendEvent(SearchBarEvent.CLICK_SEARCHBAR_OVERLAY);
+            }
+        }
+
+        this.searchBar.onclick = () => {
+            if (selectDomElement('#search-bar') === null) {
+                return;
+            }
+            this.toggleSearchBar();
         };
-        this.searchBtn.onclick = () => { 
+    }
+
+    addListenerSearchBtn() {
+        this.searchBtn.onclick = () => {
             this.sendEvent(SearchBarEvent.CLICK_SEARCH);
         };
     }
 
+    toggleSearchBar() {
+        selectDomElement('#search-input').classList.add('active-search-bar');
```

Comment:
define these elements once after view init. In your case you try to find DOM element each time when search bar is toggled